### PR TITLE
VLCMediaListPlayer: Add libvlc callbacks

### PR DESCRIPTION
**Pull request by Mikanbu**
This adds libvlc event callbacks for media_list_player.

### CODE REVIEW

**Comment by carolanitz** [comment]
can you check why circle is complaining? Also from the description and commit message I have no background what exactly is being solved and why this is needed :/ is there a ticket or somewhere more background?

File: Sources/VLCMediaListPlayer.m [hunk 1]
```diff
@@ -205,4 +273,28 @@ - (VLCRepeatMode)repeatMode
...
+#pragma mark - Delegate methods
+
+- (void)mediaListPlayerPlayed
+{
+    if ([_delegate respondsToSelector:@selector(mediaListPlayerPlayed)]) {
+        [_delegate mediaListPlayerPlayed:self];
+    }
+}
+
+- (void)mediaListPlayerNextItemSet:(VLCMedia *)media
```
Comment (carolanitz) [inline Sources/VLCMediaListPlayer.m:287]: if this is a set why is there only one media ?

File: Headers/Public/VLCMediaListPlayer.h [hunk 2]
```diff
@@ -36,6 +36,13 @@ typedef NS_ENUM(NSInteger, VLCRepeatMode) {
     VLCRepeatAllItems
 };
 
+@protocol VLCMediaListPlayerDelegate <NSObject>
+@optional
+- (void)mediaListPlayerPlayed:(VLCMediaListPlayer *)player;
```
Comment (carolanitz) [inline Headers/Public/VLCMediaListPlayer.h]: startedPlayback? or is this the callback for being done with playback. 
comments might help especially for other users of the sdk

File: Headers/Public/VLCMediaListPlayer.h [hunk 3]
```diff
@@ -36,6 +36,13 @@ typedef NS_ENUM(NSInteger, VLCRepeatMode) {
...
+- (void)mediaListPlayer:(VLCMediaListPlayer *)player nextMedia:(VLCMedia *)media;
```
Comment (carolanitz) [inline Headers/Public/VLCMediaListPlayer.h]: also here I'm unsure in what state we're in from the callback name

**Review by carolanitz — APPROVED** [review]
just add some comments update to 2019 add yourself as author and I think we're good here

**Comment by Mikanbu** [comment]
Reviews addressed. 🎉

File: Sources/VLCMediaListPlayer.m (hunk 1 above)
Comment (Mikanbu) [reply Sources/VLCMediaListPlayer.m:287]: The name is a bit confusing but this is mirroring the libvlc API.
We are exposing as `mediaListPlayer:nextMedia:` publicly.

**Review by carolanitz — APPROVED** [review]

**Comment by Mikanbu** [comment]
Rebased and merged. Thanks for the reviews! 🎉

**Pull request merged**